content/2016.09.jedit-trackers-with-kinto-en.md

```diff
@@ -0,0 +1,403 @@
+
+## updater.py
+
+There is no way to get a real-time modification stream from Sourceforge.
+
+A few ideas to work around that:
+
+ 1. poll each tracker's RSS feed (limited to 10 tickets so one can miss updates if the script lags at some point);
+ 2. poll each tracker's REST API, asking for tickets modified since last time (delay < polling period);
+ 3. subscribe to the jedit-devel mailing list and fetch mails with a pop3 client (delay < 1 minute).
+ 
+I've chosen solution 2. The [updater.py](https://github.com/elelay/jedit-trackers-kinto/blob/master/updater.py) script:
+
+ 1. fetches the trackers list from Sourceforge,
+ 2. fetches from kinto the last update date (a document in the `updater` collection),
+ 3. queries Sourceforge for tickets modified since this date,
+ 4. pushes updates to the `tickets` and `oldTickets` collections.
+ 
+Tickets are returned by Sourceforge in JSON format. There is only minimal mapping to do (fetch comments via a second
+request to Sourceforge).
+
+## Client Notifications
+
+Kinto seems to provide [websockets](http://kinto.readthedocs.io/en/latest/tutorials/notifications-websockets.html)
+notifications but I've not looked into it.
+
+Clients are instead polling the kinto server for new tickets every minute. When there are, a little *would you like to*
+*sync ?* notification appears on the top-right of the page.
+
+Following a synchronization, I update the lunr index with modified/added/deleted documents (`index.add(document)`,
+`index.remove(document)`).
+
+Counters displayed on buttons (missed, open bugs) are then recomputed and stored in a client collection.
+When there are no deletions I can update them incrementally, otherwise I have to go through the whole collection
+to recompute them.
+
+# Conclusion
+
+## Was Kinto Worth a Try ?
+
+**Definitly yes**. It's a sound platform and a good fit for my application. However it's not an out-of-box experience
+like [meteor](https://www.meteor.com/). One has to combine it with other libraries to create a full application.
+On the other hand, it makes deployment and debugging simpler.
+
+Performance is fine (batch importing the 8k tickets keeps my little server busy for 5 minutes) and the system is open.
+
+I made things a bit complicated for myself by splitting tickets into 2 collections, but not loading the old tickets
+is a huge performance boost.
+
+## To Do List (not the tutorial)
+
+I've spent 3 weeks exploring Kinto and still have things to try out. I'm also far from a full-fledged application.
+
+Some leads:
+
+**Modifying Tickets**  
+I'd like to be able to modify tickets from the application, instead of having to use the Sourceforge UI.
+Modifications could be possible offline, kept in kinto.js until network connectivity is restored.
+
+Nothing prevents it since a [little bug](https://sourceforge.net/p/forge/site-support/13403/) with
+Sourceforge API access via CORS from France has been fixed, but it take a lot of development to make a nice UI.
```

Comment:
This sentence seems a little strange to me but I'm not sure exactly why. I think it's the word "since" here.

"since" can also mean something like "because" or "now that" (Wiktionary offers the French translations « puisque » and « vu que »). If that's the meaning you're going for, I think there's some kind of conflict between "since" and "a little bug" -- it feels like "since" should introduce something definite or even final. So maybe "since the little bug" would be OK. Disclaimer: I am not an English professor, YMMV, etc. :)

On the other hand, I think you're trying to refer to a period of time (since in French, you used « depuis »). If this is what you want, I think you should put the time the bug was fixed ("Nothing has prevented this since 2008, when the little bug was fixed").